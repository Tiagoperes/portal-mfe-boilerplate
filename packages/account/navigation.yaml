+ root ~ root.account (/account):
  + environments (/environments):
    search: string
  + organization (/organization):
  + workspaces (/workspaces):
    search: string
    type: string ('ALL' | 'OWNED')
