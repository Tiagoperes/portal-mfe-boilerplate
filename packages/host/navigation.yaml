# HOME
+ root (/):
  # ACCOUNT
  + account (/account/*):
  # STUDIOS
  + studios (/studios/*):
