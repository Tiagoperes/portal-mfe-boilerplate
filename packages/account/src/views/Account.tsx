import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Account = ({ route }: ViewPropsOf<'root'>) => (
  <>
    <Breadcrumb />
    <p>Account</p>
    <a href={route.organization.$link()}>Organization</a>
    <a href={route.environments.$link({ search: 'minha busca de ambiente' })}>
      Environments
    </a>
    <a
      href={route.workspaces.$link({
        search: 'minha busca de workspace',
        type: 'ALL',
      })}
    >
      Workspaces
    </a>
  </>
)
