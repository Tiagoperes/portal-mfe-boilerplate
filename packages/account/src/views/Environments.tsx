import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Environments = ({
  route,
  params: { search },
}: ViewPropsOf<'root.environments'>) => (
  <>
    <Breadcrumb />
    <p>Environments ({search})</p>
    <a href={route.$parent.$link()}>Go back to Account</a>
  </>
)
