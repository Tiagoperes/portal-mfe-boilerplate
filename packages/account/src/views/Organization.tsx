import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Organization = ({ route }: ViewPropsOf<'root.organization'>) => (
  <>
    <Breadcrumb />
    <p>Organization</p>
    <a href={route.$parent.$link()}>Go back to Account</a>
  </>
)
