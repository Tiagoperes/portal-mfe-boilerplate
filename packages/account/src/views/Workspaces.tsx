import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Workspaces = ({
  route,
  params: { search, type },
}: ViewPropsOf<'root.workspaces'>) => (
  <>
    <Breadcrumb />
    <p>
      Workspaces ({search}, {type})
    </p>
    <a href={route.$parent.$link()}>Go back to Account</a>
  </>
)
