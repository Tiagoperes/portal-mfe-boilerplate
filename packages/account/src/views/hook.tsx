import { useNavigationContext } from 'navigation'
import { ReactElement, useState } from 'react'

export function useContent() {
  const [content, setContent] = useState<ReactElement | null>(null)

  useNavigationContext(context => {
    context
      .when('root', async props => {
        const { Account } = await import('./Account')
        setContent(<Account {...props} />)
      })
      .when('root.environments', async props => {
        const { Environments } = await import('./Environments')
        setContent(<Environments {...props} />)
      })
      .when('root.workspaces', async props => {
        const { Workspaces } = await import('./Workspaces')
        setContent(<Workspaces {...props} />)
      })
      .when('root.organization', async props => {
        const { Organization } = await import('./Organization')
        setContent(<Organization {...props} />)
      })
      .whenNotFound(() => setContent(<h1>404: Not Found</h1>))
  })

  return content
}

// Default export is required by Module Federation
// eslint-disable-next-line import/no-default-export
export default useContent
