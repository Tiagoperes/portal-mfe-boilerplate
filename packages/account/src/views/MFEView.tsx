import { useContent } from './hook'

const MFEView = () => {
  const content = useContent()
  return content
}

// Default export is required by Module Federation
// eslint-disable-next-line import/no-default-export
export default MFEView
