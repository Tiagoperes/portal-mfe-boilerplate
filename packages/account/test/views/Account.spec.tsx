import { CSSToCitricAdapter } from '@stack-spot/portal-theme'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { root } from 'navigation'
import { beforeAll, describe, expect, it } from 'vitest'
import { Account } from 'src/views/Account'

describe('Tests of Account page', () => {
  beforeAll(() => {
    root.$go()
  })
  
  it('Should match the snapshot', () => {
    const queryClient = new QueryClient()
    const { container } = render(
      <CSSToCitricAdapter>
        <QueryClientProvider client={queryClient}>
          <Account route={root} />
        </QueryClientProvider>
      </CSSToCitricAdapter>,
    )
    expect(container).toMatchSnapshot()
  })
})
