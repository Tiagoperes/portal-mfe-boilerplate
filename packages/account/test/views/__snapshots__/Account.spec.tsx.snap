// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests of Account page > Should match the snapshot 1`] = `
.c1 {
  margin: 0;
  font: 400 0.875rem/1.5 Roboto,sans-serif;
}

.c0 {
  margin-bottom: 24px;
}

.c0 ul {
  display: flex;
  flex-direction: row;
}

.c0 li {
  display: flex;
  flex-direction: row;
}

.c0 li:before {
  content: '/';
  font: tag:p;
  weight: regular;
  size: 0.875rem;
  family: system2;
  line-height: 1.5;
  margin: 0 10px;
  color: var(--light-700);
}

.c0 li:first-child:before {
  content: none;
}

.c0 a {
  color: var(--light-700);
}

.c0 a:hover {
  text-decoration: underline;
  color: var(--light-contrastText);
}

<div>
  <nav
    class="c0"
  >
    <ul>
      <li>
        <p
          class="c1"
        >
          Account
        </p>
      </li>
    </ul>
  </nav>
  <p>
    Account
  </p>
  <a
    href="/#/account/organization"
  >
    Organization
  </a>
  <a
    href="/#/account/environments?search=minha+busca+de+ambiente"
  >
    Environments
  </a>
  <a
    href="/#/account/workspaces?search=minha+busca+de+workspace&type=ALL"
  >
    Workspaces
  </a>
</div>
`;
